airflow/configuration.py

```diff
@@ -608,6 +609,48 @@ def getjson(self, section, key, fallback=_UNSET, **kwargs) -> Union[dict, list,
         except JSONDecodeError as e:
             raise AirflowConfigException(f'Unable to parse [{section}] {key!r} as valid json') from e
 
+    def gettimedelta(self, section, key, fallback=None, **kwargs) -> Optional[datetime.timedelta]:
+        """
+        Gets the config value for the given section and key, and converts it into datetime.timedelta object.
+        If the key is missing or set to a non-positive value, then it is considered as `None`.
+
+        :param section: the section from the config
+        :param key: the key defined in the given section
+        :param fallback: fallback value when no config value is given, defaults to None
+        :raises AirflowConfigException: raised because ValueError or OverflowError
+        :return: datetime.timedelta(seconds=<config_value>) or None
+        """
+        val = self.get(section, key, fallback=fallback, **kwargs)
+
+        if val:
+            # the given value must be convertible to integer
+            try:
+                int_val = int(val)
+            except ValueError:
+                raise AirflowConfigException(
+                    f'Failed to convert value to int. Please check "{key}" key in "{section}" section. '
+                    f'Current value: "{val}".'
+                )
+
+            # the given value must be positive. Otherwise fallback value is returned.
+            if int_val <= 0:
+                raise AirflowConfigException(
```

Comment:
Is this error handled anywhere, or will this cause unhandled exceptions? The config reference entry says that:
```
The missing and the non-positive values are considered as None, means that the operators are never timed out.
```

Which suggests that a missing or non-positive value shouldn't raise an error and should instead just return None.

Reply:
I've just updated the behavior of this logic based on [this comment](https://github.com/apache/airflow/pull/22389#discussion_r831289387). So, config reference has become misleading, sorry about that. Let me revisit the explanations again.

Reply:
Thanks for the warning. The description is up to date now ✅